rules-reviewed/camel3/camel2/java-generic-information.windup.groovy

```diff
@@ -1,38 +1,123 @@
-                        .perform(createMovedClassHint(MOVED_FROM, MOVED_TO, "class"))
+                        .perform(createMovedClassHint(MOVED_FROM, MOVED_TO, "class", mandatoryIssueCategory, "eips"))
         )
         .withId("java-generic-information-00034")
 
+        .addRule()
+        .when(javaMethodCallCondition("impl.SimpleRegistry.put({*})"))
+        .perform(createHint("SimpleRegistry was moved",
+                "`The org.apache.camel.support.DefaultRegistry` should be favoured instead od `SimpleRegistry`." +
+                        "Also `bind` operation should be used instead of `put` to add entries to the `SimpleRegistry` or `DefaultRegistry`.",
+                "generic_information", false)
+        )
+        .withId("java-generic-information-00035")
+
+        .addRule()
+        .when(javaMethodCallCondition("Exchange.{get|has}Out()"))
+        .perform(createHint("`getOut`/`hasOut` are deprecated",
+                "Methods `getOut`, `hasOut` on `Exchange` has been deprecated in favour of using `getMessage` instead.",
+                "getout_on_message", false))
+
+        .withId("java-generic-information-00036")
+
+        .addRule()
+        .when(Or.any(
+                javaMethodCallCondition("Message.{is|set}Fault({*})"),
+                javaMethodCallCondition("CamelContext.{is|set}HandleFault({*})")
+        ))
+        .perform(createHint("Fault API on Message was removed",
+                "Fault API was removed from `org.apache.camel.Message`. The option `handleFault` has also been removed" +
+                        " and you now need to turn this on as endpoint or component option on `camel-cxf` or `camel-spring-ws`.",
+                "fault_api_on_message", true)
+        )
+        .withId("java-generic-information-00037")
+
+        .addRule()
+        .when(Or.any(
+                javaMethodCallCondition("CamelContext.{start|stop|suspend|resume}Route({*})"),
+                javaMethodCallCondition("CamelContext.startAllRoutes()"),
+                javaMethodCallCondition("CamelContext.isStartingRoutes()"),
+                javaMethodCallCondition("CamelContext.getRouteStatus({*})")
+        ))
+        .perform(
+                createHint("Route control methods were moved",
+                        "Methods for controlling routes were moved from `CamelContext` to the `RouteController`." +
+                                "To call moved method use: `context.getRouteController().startRoute(\"myRoute\")`",
+                        "controlling_routes", true)
+        )
+        .withId("java-generic-information-00038")
+
+        .addRule()
+        .when(javaMethodCallCondition("{*}.Main.getCamelContext{s|map}()"))
+        .perform(
+                createHint("getCamelContextMap method removed",
+                        "The methods `getCamelContextMap` and `getCamelContexts` has been removed from the `Main` classes, and there is just a `getCamelContext` method now. ",
```

Comment:
"The methods `getCamelContextMap` and `getCamelContexts` has" should be "The methods `getCamelContextMap` and `getCamelContexts` have"